### DISCUSSION TITLE
ebook formats compatibility

### DISCUSSION THREAD

**Discussion by joshuachris2001** [imartinez/privateGPT#444] [Q&A]
would ebook formats such as mobi files soon be compatible? or should I just convert them in to a compatible format such as txt files.

**Comment by sime2408** [comment]
Langchain doesn't have support for the `mobi` format, that's why it was probably not included in the source code.
I usually use this software to convert mobi to epub: https://calibre-ebook.com/

or if you want to change the source code probably something like this could help (haven't test this): `pip install ebooklib`

```python
from ebooklib import epub
from ebooklib.utils import epubutils

def convert_mobi_to_epub(mobi_file, epub_file):
    # Load the MOBI file
    book = epubutils.open_book(mobi_file)

    # Create a new EPUB book
    new_book = epub.EpubBook()

    # Copy metadata from the MOBI book to the EPUB book
    new_book.set_title(book.get_title())
    new_book.set_language(book.get_language())
    new_book.set_identifier(book.get_identifier())

    # Copy each section (chapter) from the MOBI book to the EPUB book
    for section in book.get_items():
        new_section = epub.EpubHtml(title=section.get_title(), file_name=section.get_href())
        new_section.content = section.get_content()
        new_book.add_item(new_section)

    # Set the cover image, if available
    cover_image = book.get_cover()
    if cover_image:
        new_book.set_cover(file_name='cover.jpg', content=cover_image.get_content())

    # Add the table of contents
    new_book.toc = book.get_toc()

    # Define the EPUB spine (order of items)
    new_book.spine = ['nav'] + new_book.get_items()

    # Write the EPUB file
    epub.write_epub(epub_file, new_book, {})
    print("Conversion complete.")

# Usage example
convert_mobi_to_epub('input.mobi', 'output.epub')

```

**Comment by johnbrisbin** [comment]
As @sime2408 suggested, a conversion to epub would retain more of the characteristics of mobi (pagination, etc.) making it preferable to text longer term. Right now, however, epub offers no benefit over text in privateGPT because the epub reader does not add any metadata but the file name to its output and the text-splitter/chunker would not pay any attention to page data and include it in the source embeddings, since it just copies the same metadata to each embedding.
This means that all you know after a query is that some passage appeared somewhere in the source document. So unpaginated text is just as good as mobi or epub (because we have a lot of room for improvement in this area and many others).
